Tests/Algorithm/Framework/Alphas/CommonAlphaModelTests.cs

```diff
@@ -0,0 +1,265 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace QuantConnect.Tests.Algorithm.Framework.Alphas
+{
+    /// <summary>
+    /// Provides a framework for testing alpha models. 
+    /// </summary>
+    public abstract class CommonAlphaModelTests
+    {
+        private QCAlgorithmFramework _algorithm;
+        private Security _security;
+
+        [TestFixtureSetUp]
+        public void Initialize()
+        {
+            var pythonPath = new DirectoryInfo("../../../Algorithm.Framework/Alphas");
+            Environment.SetEnvironmentVariable("PYTHONPATH", pythonPath.FullName);
+
+            _algorithm = new QCAlgorithmFramework();
+            _algorithm.HistoryProvider = new SineHistoryProvider(_algorithm);
+            _algorithm.SetStartDate(2018, 1, 4);
+            _security = _algorithm.AddEquity(Symbols.SPY.Value);
+        }
+
+        [Test]
+        [TestCase(Language.CSharp)]
+        [TestCase(Language.Python)]
+        public void InsightsGenerationTest(Language language)
+        {
+            IAlphaModel model;
+            if (!TryCreateAlphaModel(language, out model))
+            {
+                Assert.Ignore($"Ignore {GetType().Name}: Could not create {language} model.");
+            }
+
+            var changes = new SecurityChanges(AddedSecurities, RemovedSecurities);
+            model.OnSecuritiesChanged(_algorithm, changes);
+
+            var allInsights = new List<Insight>();
+            var consolidators = _security.Subscriptions.SelectMany(x => x.Consolidators);
+            var slices = CreateSlices(changes);
+
+            foreach (var slice in slices)
+            {
+                var data = slice[_security.Symbol];
+                foreach (var consolidator in consolidators)
+                {
+                    consolidator.Update(data);
+                }
+                _security.SetMarketPrice(data);
+
+                var insights = model.Update(_algorithm, slice);
+                allInsights.AddRange(insights);
+            }
+
+            var actualArray = allInsights.ToArray();
+            var expectedArray = ExpectedInsights().ToArray();
+
+            Assert.AreEqual(actualArray.Length, expectedArray.Length);
+
+            for (var i = 0; i < actualArray.Length; i++)
+            {
+                var actual = actualArray[i];
+                var expected = expectedArray[i];
+                Assert.True(actual.Equals(expected));
+            }
+        }
+
+        [Test]
+        [TestCase(Language.CSharp)]
+        [TestCase(Language.Python)]
+        public void AddedSecuritiesTest(Language language)
+        {
+            IAlphaModel model;
+            if (!TryCreateAlphaModel(language, out model))
+            {
+                Assert.Ignore($"Ignore {GetType().Name}: Could not create {language} model.");
+            }
+
+            var changes = new SecurityChanges(AddedSecurities, RemovedSecurities);
```

Comment:
since we're adding `AddedSecurities` as part of initialization, I think this invocation turns into a noop. Could add another random security to the added list to ensure it turns the crank